src/client/pages/CountryHome/Repository/Panel/hooks/useOnChange.ts

```diff
@@ -15,32 +19,35 @@ export const useOnChange = (): Returned => {
   const dispatch = useAppDispatch()
   const repositoryItem = useRepositoryItem()
 
-  const { files } = useFileUploadContext()
+  const files = useUploadedFiles()
 
   const onChangeField = useCallback<OnChange>(
     (name: string, value: string) => {
-      dispatch(RepositoryActions.setRepositoryItem({ ...repositoryItem, [name]: value }))
+      dispatch(RepositoryActions.setRepositoryItemProp({ key: name as keyof RepositoryItem, value }))
     },
-    [dispatch, repositoryItem]
+    [dispatch]
   )
+
   // update repositoryItem.props
   const onChangeProps = useCallback<OnChange>(
     (name: string, value: string) => {
       const props = { ...(repositoryItem.props ?? {}), [name]: value }
-      dispatch(RepositoryActions.setRepositoryItem({ ...repositoryItem, props }))
+      dispatch(RepositoryActions.setRepositoryItemProp({ key: 'props', value: props }))
     },
     [dispatch, repositoryItem]
   )
 
-  // When a file is selected and the name is empty,
-  // set the name to the file name
   useEffect(() => {
-    if (files?.length > 0 && repositoryItem?.name === '') {
-      const file = files[0]
+    if (!files || files.length <= 0) return
+
+    const file = files[0]
+    const fileUuid = file.uuid
+    if (!Objects.isEmpty(repositoryItem) && fileUuid !== repositoryItem?.fileUuid) {
       const name = file.name.split('.')[0]
-      onChangeField('name', name)
+      dispatch(RepositoryActions.setRepositoryItemProp({ key: 'fileUuid', value: fileUuid }))
+      dispatch(RepositoryActions.setRepositoryItemProp({ key: 'name', value: name }))
```

Comment:
sure, can you elaborate a bit what u mean with partial?
what should the call look like?



> Miro, can we have setRepositoryItemProps that takes in input Partial<.... ?

you mean that we use it as `setRepositoryItemProps({ repositoryItem: { name: 'a' } })` and
`state.repositoryItem = { ...state.repositoryItem, ...payload }`

From here I understood so:

> I think it might be more efficient to have an action setRepositoryItemProp that takes only the name and value. what do you think ?